docs/middleware/BLSRegistryCoordinatorWithIndices.md

```diff
@@ -0,0 +1,80 @@
+# BLSRegistryCoordinatorWithIndices
+
+This contract is deployed for every AVS and serves as the main entrypoint for operators to register and deregister from the AVS. In addition, it is where the AVS defines its operator churn parameters.
+
+## Flows
+
+### registerOperator
+
+When registering the operator must provide 
+1. The quorums they are registering for
+2. Their BLS public key that they registered with the [BLSPubkeyCompendium](./BLSPublicKeyCompendium.md)
+3. The socket (ip:port) at which AVS offchain actors should make requests
+
+The RegistryCoordinator then
+1. Registers their BLS public key with the [BLSPubkeyRegistry](BLSPubkeyRegistry.md) and notes the hash of their public key as their operator id
+2. Registers them with the StakeRegistry 
+3. Registers them with the IndexRegistry
+4. Stores the quorum bitmap of the operator using the following struct:
+```
+/**
+ * @notice Data structure for storing info on quorum bitmap updates where the `quorumBitmap` is the bitmap of the 
+ * quorums the operator is registered for starting at (inclusive)`updateBlockNumber` and ending at (exclusive) `nextUpdateBlockNumber`
+ * @dev nextUpdateBlockNumber is initialized to 0 for the latest update
+ */
+struct QuorumBitmapUpdate {
+    uint32 updateBlockNumber;
+    uint32 nextUpdateBlockNumber;
+    uint192 quorumBitmap;
+}
+```
+
+Operators can be registered for certain quorums and later register for other (non-overlapping) quorums.
+
+### If quorum full
+
+The following struct is defined for each quorum
+```
+/**
+ * @notice Data structure for storing operator set params for a given quorum. Specifically the 
+ * `maxOperatorCount` is the maximum number of operators that can be registered for the quorum,
+ * `kickBIPsOfOperatorStake` is the basis points of a new operator needs to have of an operator they are trying to kick from the quorum,
+ * and `kickBIPsOfTotalStake` is the basis points of the total stake of the quorum that an operator needs to be below to be kicked.
```

Comment:
recommend rephrasing:
* `kickBIPsOfOperatorStake` is the multiple (in basis points) of stake that a new operator must have, as compared the operator that they are kicking out of the quorum
* and `kickBIPsOfTotalStake` is the fraction (in basis points) of the total stake of the quorum that an operator needs to be below to be kicked.

Reply:
fixed: cf883904